File: OtakuNest.OrderService/Parameters/OrderParameters.cs
using OtakuNest.Common.Parameters;
using OtakuNest.OrderService.Models;

namespace OtakuNest.OrderService.Parameters
{
    public class OrderParameters : QueryStringParameters
    {
        public string? Status { get; set; }
        public Guid? UserId { get; set; }

        public decimal? MinPrice { get; set; }
        public decimal? MaxPrice { get; set; }

        public DateTime? FromDate { get; set; }
        public DateTime? ToDate { get; set; }

        public Guid? ProductId { get; set; }
    }
}

File: OtakuNest.Client/src/app/core/params/order-parameters.ts
import { QueryStringParameters } from './query-string-parameters';

export class OrderParameters extends QueryStringParameters {
    status?: string;
    userId?: string;
    minPrice?: number;
    maxPrice?: number;
    fromDate?: string;
    toDate?: string;
    productId?: string;

    constructor() {
        super();
        this.orderBy = 'CreatedAt desc'; 
    }
}
